### ISSUE TITLE
Functions in CoreGenericFunctions.php may lose their types

### ISSUE THREAD

**Issue by bugreportuser** [vimeo/psalm#1348]
Functions in https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/Stubs/CoreGenericFunctions.php lose their types when running the PHAR with `totallyTyped="false"`.

Example:
```php
<?php
trim(array_count_values([]));
trim(array_keys([])); // false negative
```

Example output (PHAR with `totallyTyped="false"`):
```
ERROR: InvalidArgument - 2:6 - Argument 1 of trim expects string, array<array-key, int> provided
```

Example output (anything else):
```
ERROR: InvalidArgument - 2:6 - Argument 1 of trim expects string, array<array-key, int> provided
ERROR: InvalidArgument - 3:6 - Argument 1 of trim expects string, array<int, empty> provided
```

**Comment by muglug** [comment]
Um those are the same. `totallyTyped` just controls if [a set of issues](https://github.com/vimeo/psalm/blob/6b419452af934689384e0c1997fbeca730f22528/src/Psalm/Config.php#L32-L50) are reported or not

**Comment by bugreportuser** [comment]
I'm not sure I understand. The issues you linked to are only about mixed types. The problem here is that `array_keys` has an inferred return type of `mixed` when running the PHAR with `totallyTyped="false"` but a return type of `array` otherwise. I know that `totallyTyped` ignores `mixed` checks, but should it change the return type?

Plus, why would the second issue only not be emitted when running the PHAR instead of using composer?

**Comment by muglug** [comment]
Could you reproduce in a blank repo using the phar?

**Comment by bugreportuser** [comment]
Yes. I created a private repo, and you should've gotten an invite. It's only private to avoid license issues and because I plan to delete it.

**Comment by muglug** [comment]
I think these files were preserved at some point, but the phar production process was nuking the annotations cc @theofidry

**Comment by theofidry** [comment]
Annotations should be left alone, cf. [this test](https://github.com/humbug/box/blob/master/tests/Compactor/PhpTest.php#L124-L215). So if it is it's worth investigating

**Comment by weirdan** [comment]
@theofidry it appears box 3.2.0 effectively flipped the default to 'strip all (non-doctrine) annotations', when it's not specified in the config.
